lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -4433,37 +4451,51 @@ class DecomposeAtenNewEmptyStridedOp
   LogicalResult matchAndRewrite(AtenNewEmptyStridedOp op,
                                 PatternRewriter &rewriter) const override {
     SmallVector<int64_t> sizeListInts, strideListInts;
-    if (!matchPattern(op.getSize(), m_TorchListOfConstantInts(sizeListInts)))
-      return rewriter.notifyMatchFailure(
-          op, "all size list elements must be constant ints");
-    if (!matchPattern(op.getStride(),
-                      m_TorchListOfConstantInts(strideListInts)))
-      return rewriter.notifyMatchFailure(
-          op, "all stride list elements must be constant ints");
-
-    // We only support the cases with default stride values.
-    // For ex: aten.new_empty_strided(self, size=[2, 3, 4], stride=[12, 4, 1])
-    // Here the stride[0] == size[1] * size[2], stride[1] == size[2], and
-    // stride[2] == 1.
-    bool isDefaultStride = true;
-    for (unsigned i = 0; i < strideListInts.size(); i++) {
-      int64_t defaultStride = 1;
-      for (unsigned j = i + 1; j < sizeListInts.size(); j++)
-        defaultStride *= sizeListInts[j];
-      if (defaultStride != strideListInts[i]) {
-        isDefaultStride = false;
-        break;
+    if (matchPattern(op.getSize(), m_TorchListOfConstantInts(sizeListInts)) &&
+        matchPattern(op.getStride(),
+                     m_TorchListOfConstantInts(strideListInts))) {
```

Comment:
You could probably just take `op.getSize()` and `op.getStride()` directly in the helper and do the matching there. That way you don't have to duplicate this code in multiple places. You'll need to return `LogicalResult` from the helper if you do that though because it would then be possible for the helper to fail.